File: ptts/scripts/chooseFP.js
importPackage(Packages.org.csstudio.opibuilder.scriptUtil);

var macroInput = DataUtil.createMacrosInput(true);
var pv = PVUtil.getLong(pvs[0]);

macroInput.put("FP",pv);
widget.setPropertyValue("macros",macroInput);

//Reload the OPI file in the linking container again 
//by setting the property value with forcing fire option in true.
widgetController.setPropertyValue("opi_file", 
	widgetController.getPropertyValue("opi_file"), true);

File: ptts/scripts/chooseFP.py
from org.csstudio.display.builder.runtime.script import PVUtil
from org.csstudio.display.builder.runtime.script import ScriptUtil

logger = ScriptUtil.getLogger()
pv = PVUtil.getLong(pvs[0])
logger.info("PV Name " + str(pvs[0]) + " Val:"+str(pv))
widget.getPropertyValue("macros").add("FP", str(pv))
widget.setPropertyValue("file","")
widget.setPropertyValue("file","Timing_Output.bob")
